wire-schema/src/nativeMain/kotlin/com/squareup/wire/schema/SchemaLoader.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2022 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire.schema
+
+import com.squareup.wire.schema.internal.CommonSchemaLoader
+import okio.FileSystem
+
+actual class SchemaLoader : Loader, ProfileLoader {
+  private val delegate: CommonSchemaLoader
+
+  actual constructor(fileSystem: FileSystem) {
+    delegate = CommonSchemaLoader(fileSystem)
+  }
+
+  private constructor(enclosing: CommonSchemaLoader, errors: ErrorCollector) {
+    delegate = CommonSchemaLoader(enclosing, errors)
+  }
+
+  override fun withErrors(errors: ErrorCollector) = SchemaLoader(delegate, errors)
+
+  /** Strict by default. Note that golang cannot build protos with package cycles. */
+  actual var permitPackageCycles: Boolean
+    get() = delegate.permitPackageCycles
+    set(value) {
+      delegate.permitPackageCycles = value
+    }
+
+  /**
+   * If true, the schema loader will load the whole graph, including files and types not used by
+   * anything in the source path.
+   */
+  actual var loadExhaustively: Boolean
```

Comment:
Let me try in a followup

Reply:
Don't know if that's a bug or how kotlin works but even though `delegate` is a `val`, because it gets initialized in the constructor, Kotlin complains that
![image](https://github.com/square/wire/assets/1767669/a8f9658b-041f-436f-a65e-632b0f791e76)